packages/terra-list/src/ListSectionHeader.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import ListUtils from './ListUtils';
+import styles from './ListSectionHeader.module.scss';
+
+const cx = classNames.bind(styles);
+
+/* eslint-disable react/forbid-prop-types */
+const propTypes = {
+  /**
+   * Whether or not the section is collapsed.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Whether or not the section can be collapsed.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
```

Comment:
Question on the header levels and accessibility, how do these headers affect list composition when building accessibly lists? Does the content its placed within affect how a screen reader reads this content, or does the section header "break" the list reading? Also when looking at the List Subsection Header, should we remove level 1? I'm thinking with composition the highest section Header could be would be H1 and the subsection would then need to be H2?

Reply:
@neilpfeiffer @mmalaker Thoughts on this?

Reply:
Some screen readers offer the option to scroll through the individual headers on a page, and some like voiceover allow you to easily navigate to any of the headers.

<img width="764" alt="screen shot 2018-11-13 at 1 42 18 pm" src="https://user-images.githubusercontent.com/2552573/48438683-1a525680-e74a-11e8-8d1d-81140a08d32a.png">

From looking at it, there isn't any functionality specific to given headers with a 1 or a 2, but we should at least provide ways to allow consumers to define the structure they need for their app.

From looking at this, I personally favor getting rid of h1 for anything designed to be a subsection header. We will need to enforce accessibility through props, and this is one way we can ensure at least the subsections are compliant. (That is, if consumers are using it for subsections and not the actual sections)

Reply:
I would be ok with that.